utilities/workers/worker_manager.py

```diff
@@ -195,3 +218,35 @@ def join_workers(self) -> None:
         """
         for worker in self.__workers:
             worker.join()
+
+    def check_and_restart_dead_workers(self) -> None:
+        """
+        Check and restart dead workers.
+        """
+        for worker in self.__workers:
+            if not worker.is_alive():
+                # Log the error
+                frame = inspect.currentframe()
+                worker_name_string = self.__worker_properties.get_worker_name() + " worker"
+                self.__local_logger.error("Worker died, restarting " + worker_name_string, frame)
+
+                # Terminate and join worker
+                worker.terminate()
+                worker.join()
+                del worker
+                worker = None
+
+                # Create a new worker
+                result, worker = WorkerManager.__create_single_worker(
+                    self.__worker_properties.get_worker_target(),
+                    self.__worker_properties.get_worker_arguments(),
+                    self.__local_logger,
+                )
+                if not result:
+                    frame = inspect.currentframe()
+                    self.__local_logger.error("Failed to restart " + worker_name_string, frame)
```

Comment:
This should return something so that the caller knows that there is a catastrophic failure.